learn2track/models/gru_mixture.py

```diff
@@ -276,14 +280,65 @@ def _compute_losses(self, model_output):
         # means.shape : (batch_size, seq_len, n_gaussians, 3)
 
         # expected_value.shape : (batch_size, seq_len, 3)
-        expected_value = T.sum(mixture_weights[:, :, :, None] * means, axis=2)
+        self.samples = T.sum(mixture_weights[:, :, :, None] * means, axis=2)
+
+        # loss_per_time_step.shape = (batch_size, seq_len)
+        self.loss_per_time_step = T.sqrt(T.sum(((self.samples - self.dataset.symb_targets)**2), axis=2))
+        # loss_per_seq.shape = (batch_size,)
+        self.loss_per_seq = T.sum(self.loss_per_time_step*mask, axis=1) / T.sum(mask, axis=1)
+
+        return self.loss_per_seq
+
+    def _get_mixture_parameters(self, regression_output):
```

Comment:
Not needed. We should use the model's method (see related comment about a generic version of that method in the GRUMixture class).